components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -1078,21 +1079,74 @@ func (p *pg) DeleteRule(ctx context.Context, id string) error {
 		return p.processError(err)
 	}
 
-	res, err := tx.ExecContext(ctx,
-		`DELETE FROM iam_project_rules WHERE id=$1 AND projects_match_for_rule(project_id, $2);`,
-		id, pq.Array(projectsFilter),
-	)
+	assocMap, err := p.getMapOfRuleAssociations(ctx, tx, id)
 	if err != nil {
 		return p.processError(err)
 	}
 
-	count, err := res.RowsAffected()
-	if err != nil {
-		return p.processError(err)
-	} else if count == 0 {
+	_, ruleStaged := assocMap[pgStaged]
```

Comment:
The map returns `valueBasedOnKey, wasKeyFound`, but as @srenatus pointed out, since it's a map of bools, I can just look at the first value so this becomes:

```
ruleStaged := assocMap[pgStaged]
```